GWMiniGame/GWMiniGame.cpp

```diff
@@ -0,0 +1,287 @@
+	for (int i = 0; i < maxWordLength; ++i) {
+		if (array[i] == currentChar) {
+			indexDB[i] = i;
+		}
+		else {
+			indexDB[i] = falseValue;
+		}
+	}
+	for (int i = 0; i < maxWordLength; ++i) {
+		if (arrayInput[i] == currentChar) {
+			indexInput[i] = i;
+		}
+		else {
+			indexInput[i] = falseValue;
+		}
+	}
+	for (int i = 0; i < maxWordLength; ++i) {
+		if (indexDB[i] != falseValue) {
+			for (int j = 0; j < maxWordLength; ++j) {
+				if (indexDB[i] == indexInput[j]) {
+					result = true;
+					return result;
+				}
+			}
+		}
+	}
+	return result;
+}
+
+bool isWordGuessed(const std::string& arrayDB, const char* arrayInput) {
+	for (int i = 0; i < arrayDB.length() && arrayInput[i] != '\0'; ++i) {
+		char lowerCaseDB = std::tolower(arrayDB[i]);
+		char lowerCaseInput = std::tolower(arrayInput[i]);
+		if (lowerCaseDB != lowerCaseInput) {
+			return false;
+		}
+	}
+	return true;
+}
+
+char* wordCompare(const std::string& arrayDB, const char* arrayInput) {
+	const int maxWordLength = 5;
+	char countCheck[maxWordLength + 1];
+	char* result = new char[maxWordLength + 1];
+	
+	for (int i = 0; i < arrayDB.length(); ++i) {
+		char lowerCaseDB = std::tolower(arrayDB[i]);
+		char lowerCaseInput = std::tolower(arrayInput[i]);
+
+		if (lowerCaseDB == lowerCaseInput) {
+			countCheck[i] = lowerCaseInput;
+			result[i] = (char)std::toupper(lowerCaseInput);
+		}
+		else {
+			countCheck[i] = lowerCaseInput;
+			if (letterCount(countCheck, lowerCaseInput) <= letterCount(arrayDB, lowerCaseInput) && letterCount(arrayDB, lowerCaseInput) == 1) {
+				if (findIndex(arrayDB, arrayInput, lowerCaseInput)) {
+					result[i] = '*';
+				}
+				else {
+					result[i] = lowerCaseInput;
+				}
+			}
+			else if(letterCount(countCheck, lowerCaseInput)<= letterCount(arrayDB, lowerCaseInput) && letterCount(arrayDB, lowerCaseInput)>1){
+				result[i] = lowerCaseInput;
+			}
+			else {
+				result[i] = '*';
+			}
+		}
+	}
+	result[5] = '\0';
+	return result;
+}
+
+int main(){
+	while (true) {
+		int value = mainMenu();
+		if (value == 0) {
+			std::cout << "See You next time!" << std::endl;
+			break;
+		}
+		if (value == 1) {
+		const std::string& randomWord = randomWordFromBD("WordDB.txt");
+		int attemptCount = 0;
+			while (true) {
+				attemptCount++;
+				if (attemptCount == 1) {
+					std::cout << "RESULT: *****" << std::endl;
+				}
+				char* userInput = wordInput();
+				
+				if (isWordGuessed(randomWord.c_str(), userInput) == true) {
+					std::cout<< "That's right!" << std::endl;
+					std::cout << "You made " << attemptCount << " tries!" << std::endl;
+					std::cout << std::endl;
+					delete[] userInput;
+					break;
+				}
+				else {
+					std::cout << "RESULT: " << wordCompare(randomWord.c_str(), userInput) << std::endl;
+				}
+				delete[] userInput;
+			}
+		}
+		if (value == 2) {
+			if (checkDate("WordOfDay.txt")) {
+				const std::string& randomWord = randomWordFromBD("WordDB.txt");
+				int attemptCount = 0;
+				while (true) {
```

Comment:
There are to many levels of nesting in your main. It's worth to split it even more into functions. At least every menu choice can be easily its own function. And part of it can be splitted further as well